rosbag2_tests/test/rosbag2_tests/test_rosbag2_record_end_to_end.cpp

```diff
@@ -414,6 +414,72 @@ TEST_F(RecordFixture, record_end_to_end_with_splitting_splits_bagfile) {
   }
 }
 
+TEST_F(RecordFixture, record_end_to_end_with_duration_splitting_splits_bagfile) {
+  constexpr const char topic_name[] = "/test_topic";
+  constexpr const int bagfile_split_duration = 1;   // 1 second
+
+  std::stringstream command;
+  command << "ros2 bag record" <<
+    " --output " << root_bag_path_.string() <<
+    " -d " << bagfile_split_duration <<
+    " " << topic_name;
+  auto process_handle = start_execution(command.str());
+  wait_for_db();
+
+  constexpr const int expected_splits = 4;
+  {
+    constexpr const char message_str[] = "Test";
+    constexpr const int message_size = 1024 * 1024;  // 1MB
+    constexpr const int message_time = 500;  // 500ms
+    // string message from test_msgs
+    const auto message = create_string_message(message_str, message_size);
+    constexpr const int message_count = bagfile_split_duration * expected_splits / message_time;
+
+    pub_man_.run_scoped_publisher(
+      topic_name,
+      message,
+      500ms,
+      message_count);
+  }
+
+  stop_execution(process_handle);
+
+  rosbag2_storage::MetadataIo metadata_io;
+
+// TODO(zmichaels11): Remove when stop_execution properly SIGINT on Windows.
+// This is required since stop_execution hard kills the proces on Windows,
+// which prevents the metadata from being written.
+#ifdef _WIN32
+  {
+    rosbag2_storage::BagMetadata metadata;
+    metadata.version = 4;
+    metadata.storage_identifier = "sqlite3";
+
+    for (int i = 0; i < expected_splits; ++i) {
+      const auto bag_file_path = get_bag_file_name(i);
+
+      // There is no guarantee that the bagfile split expected_split times
+      // due to possible io sync delays. Instead, assert that the bagfile split
+      // at least once
+      if (rcpputils::fs::path(bag_file_path).exists()) {
+        metadata.relative_file_paths.push_back(bag_file_path);
+      }
+    }
+
+    ASSERT_GE(metadata.relative_file_paths.size(), 1) << "Bagfile never split!";
```

Comment:
This is all copy-paste from another test, but those tests aren't running that reliably so I'm not sure we should be following in their footsteps.

Regardless, should this asserting about the metadata contents be done outside the `ifdef` block? Is the condition something that should be true no matter what? It seems like it is currently _only_ testing "does vector.push_back work?", unless I'm misreading it.

Reply:
Didn't know those tests were unreliable. I'm a bit new to the testing scene, so is there a different test I should be following as a guideline?

Reply:
And judging by the TODO comments, I think it's testing if the metadata got written in Windows. Not sure why Windows would prevent metadata from being written, though, but I suppose it's enough of a problem that they needed to write a case for it.